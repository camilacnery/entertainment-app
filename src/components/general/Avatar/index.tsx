import { FC } from "react";
import Image from "next/future/image";
import styles from "./index.module.scss";

type TAvatar = {
  imageUrl: string;
  size?: number;
};

const Avatar: FC<TAvatar> = ({ imageUrl, size = 100 }) => {
  return <Image className={styles.avatar} src={imageUrl} alt={""} width={size} height={size} />;
};

export default Avatar;
