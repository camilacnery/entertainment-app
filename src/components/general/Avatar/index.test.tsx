import { render } from "@testing-library/react";

import Avatar from ".";

describe("Avatar", () => {
  it("renders component", () => {
    const { baseElement } = render(<Avatar imageUrl="https://image.url" />);

    expect(baseElement).toMatchSnapshot();
  });
});
