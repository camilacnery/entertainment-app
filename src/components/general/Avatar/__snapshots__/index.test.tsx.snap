// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders component 1`] = `
<body>
  <div>
    <img
      alt=""
      class="avatar"
      data-nimg="future"
      decoding="async"
      height="100"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fimage.url&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimage.url&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.url&w=256&q=75 2x"
      style="color: transparent;"
      width="100"
    />
  </div>
</body>
`;
